app/models/frontend/nics/NICsAnswers.scala

```diff
@@ -17,24 +17,81 @@
+    val class2Data: Option[Boolean]              = maybeDisclosures.flatMap(_.class2Nics.flatMap(_.class2VoluntaryContributions))
+    val class2DatabaseData: Option[Boolean]      = maybeDbAnswers.flatMap(_.class2NICs)
+    val class4Data: List[Class4ExemptionAnswers] = mkClass4ExemptionData(maybeAnnualSummaries)
+    (class2Data, class2DatabaseData, class4Data) match {
+      case (_, _, class4Data) if class4Data.nonEmpty => mkPriorClass4Data(class4Data, maybeDbAnswers)
+      case (Some(true), _, _)                        => Some(NICsAnswers(class2Answers = NICsClass2Answers(true).some, None))
+      case (None, Some(false), _)                    => Some(NICsAnswers(class2Answers = NICsClass2Answers(false).some, None))
+      case _                                         => None
+    }
+  }
+
+  def mkClass4ExemptionData(multipleAnnualSummaries: List[(api_1803.SuccessResponseSchema, BusinessId)]): List[Class4ExemptionAnswers] =
+    multipleAnnualSummaries.flatMap { case (answers, businessId) =>
+      val maybeClass4Exemption: Option[Boolean] = answers.annualNonFinancials.flatMap(_.exemptFromPayingClass4Nics)
+      maybeClass4Exemption.map { class4Exempt =>
+        val class4ExemptReason: Option[ExemptionReason] = {
+          val reason: Option[AnnualNonFinancialsType.Class4NicsExemptionReason.Value] =
+            answers.annualNonFinancials.flatMap(_.class4NicsExemptionReason)
+          reason.map(ExemptionReason.fromNonFinancialType)
+        }
+        Class4ExemptionAnswers(businessId, class4Exempt, class4ExemptReason)
+      }
+    }
+
+  def mkPriorClass4Data(class4Answers: List[Class4ExemptionAnswers], maybeDbAnswers: Option[NICsStorageAnswers]): Option[NICsAnswers] = {
+    val journeyIsYesButNoneAreExempt = maybeDbAnswers.exists(_.journeyIsYesButNoneAreExempt.contains(true))
+
+    if (class4Answers.length > 1) buildMultipleBusinessAnswers(class4Answers, journeyIsYesButNoneAreExempt)
+    else buildSingleBusinessAnswers(class4Answers)
+  }
+
+  private def buildSingleBusinessAnswers(class4Answers: List[Class4ExemptionAnswers]): Option[NICsAnswers] = class4Answers.headOption.map {
+    singleExemption =>
+      val singleClass4Answers = NICsClass4Answers(singleExemption.class4Exempt, singleExemption.exemptionReason, None, None)
+      NICsAnswers(None, Some(singleClass4Answers))
+  }
+
+  private def buildMultipleBusinessAnswers(class4Answers: List[Class4ExemptionAnswers],
+                                           journeyIsYesButNoneAreExempt: Boolean): Option[NICsAnswers] = {
+    val class4NicsBoolean        = class4Answers.exists(_.class4Exempt)
+    val class4DivingBusinessIds  = returnBusinessIdsForRelevantExemption(class4Answers, ExemptionReason.DiverDivingInstructor)
+    val class4TrusteeBusinessIds = returnBusinessIdsForRelevantExemption(class4Answers, ExemptionReason.TrusteeExecutorAdmin)
+
```

Comment:
It is possible for a user to say 'Yes' they are class 4 exempt, 'Exempt for other reason' on diving exemption page, and then 'None of these businesses are exempt' on non-diving exemption page.
In this case, we will 'false' for class 4 exemption to the api and save this 'journeyIsYesButNoneAreExempt' value in the database.

However, if they have no session data then we need to restore their (contradictory) 'yes but no exemptions' answers rather than the 'false/No' answers returned from the API.
The first case below handles this.
If the diving exemptions are empty then none will be returned from api but we need to populate the CYA page with the 'Exempt for other reason' message (handled by case 2).
If none for non-diving then that row is not shown on CYA.